sympy/polys/domains/mpelements.py

```diff
@@ -22,10 +22,7 @@ def _set_mpf(self, val):
         prec, rounding = self.context._prec_rounding
         tol = self.context.tol
 
-        if mpf_lt(mpf_abs(val, prec, rounding), tol):
-            self.__mpf__ = fzero
-        else:
-            self.__mpf__ = val
+        self.__mpf__ = val
```

Comment:
So as seen in this PR, sometimes you want it to go to zero and sometimes you don't. In particular, a floating point number, let's say double precision, can hold 15 digits, but the mantissa is arbitrary (for machine doubles, the mantissa integer must fit within 11 bits, for mpf I think there is no limitation). As such, the precision is relative precision. You can easily represent numbers like 1.2345e-100. No problem:
```
In [1]: 1.2345678901234567e-100
Out[1]: 1.2345678901234567e-100

In [2]: _ * 2
Out[2]: 2.4691357802469134e-100

In [3]: _ / 2
Out[3]: 1.2345678901234567e-100

In [4]: _ * 3
Out[4]: 3.70370367037037e-100

In [5]: _ / 3
Out[5]: 1.2345678901234567e-100
```
It holds ~15 digits. You do not want this to become 0, only because the number is smaller than 1e-16:
```
In [6]: 1.2345678901234567e-100 < 1e-16
Out[6]: True
```
and neither we want in sympy, because then it loses the digits as reported in #9398, which this PR fixes.

In applications where you really want this behavior, then you just need to "chop" these small numbers in higher level code. As you can see, all tests pass in sympy with this PR, so we do not actually need this feature (yet). But if we do, you just chop it.

On the other hand, if you always chop it, then you are losing precision and you can't recover it later once you chop it. 

For that reason, it should not be chopped at this low level, but only at higher level, if needed.